judgments/tests/test_atom_feed.py

```diff
@@ -45,11 +48,54 @@ def test_feed_exists(self, mock_api_client, mock_search_judgments_and_parse_resp
         self.assertIn("<author><name>court</name></author>", decoded_response)
 
     @patch("judgments.feeds.search_judgments_and_parse_response")
-    def test_bad_page_404(self, mock_search_judgments_and_parse_response):
-        # "?page=" 404s, not 500
+    @patch("judgments.feeds.api_client")
+    def test_search_query_in_URL(self, mock_api_client, mock_search_judgments_and_parse_response):
+        search_response = FakeSearchResponseManyPages()
+        mock_search_judgments_and_parse_response.return_value = search_response
+
+        response = self.client.get("/atom.xml?query=obscure-search-query&page=5&order=date")
+        decoded_response = response.content.decode("utf-8")
+
+        # that search_judgments_and_parse_response is called with the appropriate parameters
+        mock_search_judgments_and_parse_response.assert_called_with(
+            mock_api_client,
+            SearchParameters(
+                query="obscure-search-query",
+                court="",
+                judge=None,
+                party=None,
+                date_from="2003-01-01",
+                date_to=None,
+                order="date",
+                page=5,
+            ),
+        )
+
+        self.assertIn("Search results for obscure-search-query", decoded_response)
+
+        self.assertIn(
+            '"https://caselaw.nationalarchives.gov.uk/atom.xml?query=obscure-search-query&amp;order=date"',
+            decoded_response,
+        )
+        self.assertIn(
+            '"https://caselaw.nationalarchives.gov.uk/atom.xml?query=obscure-search-query&amp;order=date&amp;page=4"',
+            decoded_response,
+        )
+        self.assertIn(
+            '"https://caselaw.nationalarchives.gov.uk/atom.xml?query=obscure-search-query&amp;order=date&amp;page=6"',
+            decoded_response,
+        )
+        self.assertIn(
+            '"https://caselaw.nationalarchives.gov.uk/atom.xml?query=obscure-search-query&amp;order=date&amp;page=100"',
+            decoded_response,
+        )
+
+    @patch("judgments.feeds.search_judgments_and_parse_response")
+    def test_bad_page_not_500(self, mock_search_judgments_and_parse_response):
+        # "?page=" doesn't 500
         mock_search_judgments_and_parse_response.return_value = FakeSearchResponse()
         response = self.client.get("/atom.xml?page=")
-        self.assertEqual(response.status_code, 404)
+        assert response.status_code != 500
```

Comment:
Shouldn't this really be a 400? The user has provided an invalid parameter (`page` is `None`)

Reply:
It's not invalid to not specify a page; due to changes in the handling, an empty page is now handled as 1, so we get a 200. I could explicitly test for that.

Reply:
We should probably make this match the spec as much as possible, so if we could explicitly require page (tbh any parameter) to be of the expected type that's probably preferable...

But also I don't feel massively strongly here. This just feels like a non-test; what case is it actually testing?

Reply:
I think this test was added to avoid regressions of a bug where users could provoke 500 errors. But yeah, we can ditch it.

Reply:
We deleted it.